.github/workflows/my_ci_pipeline.yaml

```
@@ -0,0 +1,16 @@
+name: Continuous Integration
+on: 
+  push:
+    paths-ignore:
+      - 'README.md'
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```

Minor (and I know we're out of date in some places!) but checkout is actually on v4 now which should fix up some of your warnings:
```suggestion
    - uses: actions/checkout@v4
```